rust/private/rustc.bzl

```diff
@@ -584,6 +584,41 @@ def collect_inputs(
 
     return compile_inputs, out_dir, build_env_files, build_flags_files, linkstamp_outs, ambiguous_libs
 
+def _collect_ld_library_path(env, rust_toolchain, cc_toolchain):
+    """Construct the value of `LD_LIBRARY_PATH` based on the existing environment and registered toolchains.
+
+    Args:
+        env (dict): A dict of existing environment variables
+        rust_toolchain (rust_toolchain): The registered rust_toolchain
+        cc_toolchain (CcToolchainInfo): A cc toolchain provider to be used.
+
+    Returns:
+        str: The value of `LD_LIBRARY_PATH`.
+    """
+    sep = ";"
+    ld_paths = []
+
+    # Maintain existing values
+    if "LD_LIBRARY_PATH" in env:
+        ld_paths.extend(env["LD_LIBRARY_PATH"].split(sep))
+    if "DYLD_LIBRARY_PATH" in env:
```

Comment:
This is mac-only, right? I guess we shouldn't set the variable on Windows or Linux.

Reply:
> Then IMHO it's the job of the toolchain to prepare the correct environment variables, not rustc.bzl. I'll comment on the code to be more precise in a second.

I think I agree. Would it make sense to allow `rust_toolchain` to set environment variables? Or how could I have the toolchain define this information?

Reply:
Yeah, unfortunately we don't really have a consistent way to detect the exec environment since toolchains are not required to provide a triple.

Reply:
And actually this is setting the variable only when it's already present. I think that it's OK in that case, sorry for not realizing this earlier.